keras_nlp/layers/transformer_decoder.py

```diff
@@ -165,7 +165,7 @@ def _build(self, input_shape, has_cross_attention):
             self._cross_attention_layer = keras.layers.MultiHeadAttention(
                 num_heads=self.num_heads,
                 key_dim=head_dim,
-                value_dim=hidden_dim,
+                value_dim=head_dim,
```

Comment:
Whoa, did you find a bug here? If so, let's discuss and break this into a different PR.

Reply:
Done!